web/src/emoji_picker.js

```diff
@@ -601,6 +611,34 @@ function register_popover_events($popover) {
 
     $(".emoji-popover-filter").on("input", filter_emojis);
     $(".emoji-popover-filter").on("keydown", process_enter_while_filtering);
+    $(".emoji-popover-filter").on("keydown", () => {
+        const $search_input = $("#emoji-search-query");
+        const $clear_button = $("#emoji_search_clear");
+        $(document).on("input", "#emoji-search-query", () => {
+            if ($search_input.val() === "") {
+                $clear_button.hide();
+            } else {
+                $clear_button.show();
+            }
+        });
+
+        $(document).on("click", "#emoji_search_clear", async (e) => {
```

Comment:
Why are we using `async` over here? I don't see any I/O-bound or network-bound requests.